dbms/src/Server/Client.cpp

```
@@ -550,9 +551,33 @@ class Client : public Poco::Util::Application
             while (begin < end)
             {
                 const char * pos = begin;
-                ASTPtr ast = parseQuery(pos, end, true);
+                ASTPtr ast = nullptr;
+                try
+                {
+                    ast = parseQuery(pos, end, true);
+                }
+                catch (...)
+                {
+                    if (!ignore_error)
+                        throw;
+                }
```

The error is silently swallowed in case of `ignore_error` - unacceptable.